reframe/core/config.py

```diff
@@ -217,3 +218,106 @@ def create_env(system, partition, name):
                 system.add_partition(part)
 
             self._systems[sys_name] = system
+
+
+def create_json_object():
```

Comment:
The interface I want is this one:

```python
def convert_old_config(filename: str) -> str:
```
This should read the file `filename` convert it to the new file and store it in a temporary file, whose name it will return.